### DISCUSSION TITLE
Passing props to react components

### DISCUSSION THREAD

**Discussion by meeoh** [ElMassimo/vite_ruby#313] [Q&A]
Hey, loving vite_ruby so far. Using it with React. Is there a way to pass data to my client side comopnents on the initial request? E.g doing something like:
` <%= vite_typescript_tag 'application.tsx' user={current_user} %>`

My alternative is that when someone visits my application, i'll first have to make a request to my API layer and see if the users logged in and display info based on the result of that call. 

I would assume theres some way to do this similar to the rails/webpacker approach

**Comment by ElMassimo** [comment]
Hi @meeoh!

[Inertia.js](https://inertiajs.com/) is a great solution for this.

Highly recommend reading their docs, but in summary, the first request to the server allows you to return the data that your client-side app will use, and will pass it as props to your page components.

---

If you are looking for something lighter, you can pick up a few tricks from Inertia, like [serializing your JSON data into your HTML using a `data` tag](https://github.com/inertiajs/inertia-rails/blob/master/app/views/inertia.html.erb#L1), and then [retrieving and parsing that data](https://github.com/inertiajs/inertia/blob/5298add1abb76ab2478cd3f5df4753fcc9b9156d/packages/react/src/createInertiaApp.js#L8) before mounting your React app.

I'd definitely [try Inertia](https://github.com/inertiajs/inertia-rails) first though 😃

**Reply by meeoh** [reply]
Thanks so much, never heard of Inertia.js, will definitely integrate it. Also really appreciate the fast responses @ElMassimo
